fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1509,6 +1509,144 @@ void simpleCrossPartitionQuery() {
         }
     }
 
+    static Stream<Arguments> findStartingPartitionTest() {
+        return Stream.of(true, false).map(Arguments::of);
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void findStartingPartitionTest(boolean isSynthetic) {
+
+        final Map<String, String> options = Map.of(
+                INDEX_PARTITION_BY_FIELD_NAME, isSynthetic ? "complex.timestamp" : "timestamp",
+                INDEX_PARTITION_HIGH_WATERMARK, String.valueOf(8));
+        Pair<Index, Consumer<FDBRecordContext>> indexConsumerPair = setupIndex(options, true, false);
+        final Index index = indexConsumerPair.getLeft();
+        Consumer<FDBRecordContext> schemaSetup = indexConsumerPair.getRight();
+
+        final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_REPARTITION_DOCUMENT_COUNT, 8)
+                .build();
+        final String luceneSearch = "text:about";
+
+        try (FDBRecordContext context = openContext(contextProps)) {
+            schemaSetup.accept(context);
+            LucenePartitioner partitioner = getIndexMaintainer(index).getPartitioner();
+            Tuple groupKey = Tuple.from(1L);
+            
+            // timestamps present in partition keys
+            long time0 = System.currentTimeMillis();
+            long time1 = System.currentTimeMillis() + 1000;
+            long time2 = System.currentTimeMillis() + 2000;
+            
+            // timestamp not in partition keys, but within a partition
+            long time1_2 = time1 + 500; // between time1 and time2
+
+            // timestamps outside the bounds of the partitions
+            long timeTooOld = time0 - 500;
+            long timeTooNew = time2 + 500;
+
+            Map<Long, String> timesForLogging = Map.of(
+                    time0, "time0",
+                    time1, "time1",
+                    time2, "time2",
+                    time1_2, "time1_2",
+                    timeTooOld, "timeTooOld",
+                    timeTooNew, "timeTooNew");
+
+            //
+            // p0: from: t0+1200 to: t0+1300
+            // p1: from: t0+1301 to: t1+1400
+            // p2: from: t1+1410 to: t2+1500
+            // p3: from: t2+1510 to: t2+1600
+            // create partition metadata directly rather than depending on repartitioning to ensure that the partitions
+            createPartitionMetadata(index, groupKey, 0, time0, time0, Tuple.from(1, 1200), Tuple.from(1, 1300));
+            createPartitionMetadata(index, groupKey, 1, time0, time1, Tuple.from(1, 1301), Tuple.from(1, 1400));
```

Comment:
It would be good to have a test where there is a gap between the `to` of one partition and the `from` of the next partition